File: backend/services/user_service.py
from sqlalchemy.orm import Session
from fastapi import HTTPException, status

from core.security import hash_password
from models.user import User
from schemas.user import CreateUserRequest


def get_users(organization_id: int, db: Session) -> list[User]:
    return db.query(User).filter(User.organization_id == organization_id).all()


def create_user(request: CreateUserRequest, organization_id: int, db: Session) -> User:
    if db.query(User).filter(User.email == request.email).first():
        raise HTTPException(
            status_code=status.HTTP_409_CONFLICT,
            detail="Email already exists",
        )

    user = User(
        organization_id=organization_id,
        department_id=request.department_id,
        name=request.name,
        email=request.email,
        password_hash=hash_password(request.password),
        role=request.role,
    )
    db.add(user)
    db.commit()
    db.refresh(user)
    return user

File: frontend/src/app/core/services/user.service.ts
import { Injectable, inject } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';

import { environment } from '../../../environments/environment';
import { User } from '../models/user.model';

export interface CreateUserRequest {
  name: string;
  email: string;
  password: string;
  role: string;
  department_id: number | null;
}

@Injectable({ providedIn: 'root' })
export class UserService {
  private readonly http = inject(HttpClient);

  getUsers(): Observable<User[]> {
    return this.http.get<User[]>(`${environment.apiUrl}/users`);
  }

  createUser(request: CreateUserRequest): Observable<User> {
    return this.http.post<User>(`${environment.apiUrl}/users`, request);
  }
}
